test/csuite/timestamp_abort/main.c

```diff
@@ -148,10 +148,11 @@ thread_ts_run(void *arg)
 			 * any thread still with a zero timestamp we go to
 			 * sleep.
 			 */
-			if (th_ts[i] == 0)
+			this_ts = th_ts[i];
+			if (this_ts == 0)
 				goto ts_wait;
-			if (th_ts[i] != 0 && th_ts[i] < oldest_ts)
-				oldest_ts = th_ts[i];
+			if (this_ts != 0 && this_ts < oldest_ts)
```

Comment:
I don't think there is any need to re-check `this_ts` for 0 here, is there?

I'm concerned that the compiler will just get rid of the local variable.

I also don't really see how re-reading `this_ts` could lead to oldest moving further forward than expected. This code isn't relying on seeing a stable view of the values and once it's published the thread has finished committing their transaction. Could you describe the scenario that triggers a failure?

Reply:
I put the description in the ticket itself. I think you're right that there's no need to recheck for 0.  I'll fix that.